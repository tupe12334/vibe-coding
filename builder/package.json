{
  "name": "@vibe-builder/builder",
  "version": "0.1.0",
  "description": "Generate Vibe coding agent instructions",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "bin": {
    "vibe-builder": "dist/cli/main.js"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vibe-code/vibe-coding.git"
  },
  "scripts": {
    "build": "tsc",
    "prebuild": "rimraf dist",
    "build:w": "tsc --watch",
    "start:cli": "node dist/cli/main.js",
    "test": "vitest run",
    "test:watch": "vitest watch",
    "test:ui": "vitest --ui",
    "test:update": "vitest -u",
    "test:watch:update": "vitest watch -u"
  },
  "devDependencies": {
    "@types/mdast": "^4.0.4",
    "@types/node": "^24.0.1",
    "@vitest/ui": "3.2.3",
    "rimraf": "^6.0.1",
    "vitest": "^3.2.3"
  },
  "dependencies": {
    "mdast-util-to-markdown": "^2.1.2"
  }
}
