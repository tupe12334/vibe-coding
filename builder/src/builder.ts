export interface BuilderOptions {
  language: string;
}
export function builder(options: BuilderOptions) {
    
}
