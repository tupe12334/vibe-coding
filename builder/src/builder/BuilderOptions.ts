export type BuilderOptions =
  | {
      language: string;
      projectType?: string;
      packageManager?: string;
    }
  | {
      language?: never;
      projectType?: string;
      packageManager?: never;
    };
