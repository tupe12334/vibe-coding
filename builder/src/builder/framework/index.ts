export { frameworkSegment } from "./frameworkSegment";
export { Frameworks, getAvailableFrameworks } from "./options";
export type { FrameworkOption } from "./options";
