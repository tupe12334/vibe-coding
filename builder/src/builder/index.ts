export * from "./builder";
export { BuilderOptions } from "./BuilderOptions";
export { Languages } from "./language";
export { Frameworks, getAvailableFrameworks } from "./framework";
