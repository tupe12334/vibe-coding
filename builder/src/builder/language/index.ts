export { Languages } from "./options";
export { languageSegment } from "./languageSegment";
