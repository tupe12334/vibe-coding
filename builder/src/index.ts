export * from "./builder";
export * from "./cli";
