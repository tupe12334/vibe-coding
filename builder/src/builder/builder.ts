import { Root } from "mdast";
import { toMarkdown } from "mdast-util-to-markdown";
import { join } from "path";
import { languageSegment } from "./language/languageSegment";

export interface BuilderOptions {
  language: string;
}
export async function builder(options: BuilderOptions) {
  const tree: Root = {
    type: "root",
    children: [
      {
        type: "heading",
        depth: 1,
        children: [{ type: "text", value: "AI agent instructions" }],
      },
    ],
  };
  const templatesPath = join(__dirname, "../../templates");

  tree.children = tree.children.concat(
    await languageSegment(templatesPath, options.language)
  );

  return toMarkdown(tree);
}
