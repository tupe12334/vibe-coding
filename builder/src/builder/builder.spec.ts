import { describe, expect, test } from "vitest";
import { builder } from "./builder";
describe("builder", () => {
  test("working with no arguments returns general segment only", async () => {
    const response = await builder();

    expect(response).toMatchSnapshot();
  });

  test("working with language only", async () => {
    const response = await builder({ language: "typescript" });

    expect(response).toMatchSnapshot();
  });
  test("working with language and package manager", async () => {
    const response = await builder({
      language: "typescript",
      packageManager: "npm",
    });

    expect(response).toMatchSnapshot();
  });
  test("working with project type only", async () => {
    const response = await builder({ projectType: "frontend" });

    expect(response).toMatchSnapshot();
  });
});
