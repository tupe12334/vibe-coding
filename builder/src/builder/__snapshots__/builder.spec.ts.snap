// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`builder > working with language and package manager 1`] = `
"# AI agent instructions

## General Guidelines

### Do

* Always try to early return from functions.

* Use \`const\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

Package Manager: npm

## Programming language (typescript)

### Do

* Use Interfaces over Types unless needed.

### Don't
"
`;

exports[`builder > working with language only 1`] = `
"# AI agent instructions

## General Guidelines

### Do

* Always try to early return from functions.

* Use \`const\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

## Programming language (typescript)

### Do

* Use Interfaces over Types unless needed.

### Don't
"
`;

exports[`builder > working with no arguments returns general segment only 1`] = `
"# AI agent instructions

## General Guidelines

### Do

* Always try to early return from functions.

* Use \`const\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.
"
`;

exports[`builder > working with project type only 1`] = `
"# AI agent instructions

## General Guidelines

### Do

* Always try to early return from functions.

* Use \`const\` for variables that are not reassigned.

* Make sure to focus on why and not how in documentation.

## Project type (frontend)

### Do

* Prefer working with flex layouts for responsive design.

* Use rems for font sizes and spacing to ensure scalability.
"
`;
